src/rustup-cli/proxy_mode.rs

```diff
@@ -26,14 +26,15 @@ pub fn main() -> Result<()> {
 
         // Check for a toolchain specifier.
         let arg1 = args.next();
-        let toolchain = arg1.as_ref().and_then(|arg1| {
-            if arg1.starts_with('+') {
-                Some(&arg1[1..])
-            } else {
-                None
-            }
-        });
-
+        let toolchain = arg1.as_ref()
+            .and_then(|arg1| arg1.to_str())
+            .and_then(|arg1| {
+                if arg1.starts_with('+') {
+                    Some(&arg1[1..])
+                } else {
+                    None
+                }
+            });
```

Comment:
Could we do the `.starts_with('+')` before the `.to_str()` and issue an error if the user has specified a non-utf8 toolchain name?